File: GARI_Umbria_2026-08-03/GiasNG/src/app/Model/attivita/dettagli/AvversitaTrappole.ts
import {BaseCodeDescr} from "../../baseClass/baseCodeDescr";
import {AvversitaGruppo} from "../../metaschema/avversita/AvversitaGruppo";
import {UnitaDiMisura} from "../../metaschema/UnitaDiMisura";
import {UtilizzoAvversitaTrappole} from "./UtilizzoAvversitaTrappole";

export class AvversitaTrappole extends BaseCodeDescr{
  public avversitaGruppo: AvversitaGruppo;
  public unitaDiMisura: UnitaDiMisura;
  public utilizzi: UtilizzoAvversitaTrappole[];

  constructor(codice: number, descrizione: string) {
    super(codice, descrizione);

  }
}

File: GARI_Umbria_2026-08-03/DotNet/AgronicaCoreStd/AgronicaCoreModelsSTD/attivita/dettagli/AvversitaTrappole.cs
using AgronicaCoreModelsSTD.baseClass;
using AgronicaCoreModelsSTD.metaschema;
using AgronicaCoreModelsSTD.metaschema.avversita;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace AgronicaCoreModelsSTD.attivita.dettagli
{
    public class AvversitaTrappole: BaseCodeDescr
    {
        public AvversitaGruppo avversitaGruppo { get; set; }

        public UnitaDiMisura unitaDiMisura { get; set; }

        public List<UtilizzoAvversitaTrappole> utilizzi { get; set; }

        public new AvversitaTrappole Clona()
        {
            try
            {

                string output = JsonConvert.SerializeObject(this);
                AvversitaTrappole deserializedObject = JsonConvert.DeserializeObject<AvversitaTrappole>(output);
                return deserializedObject;
            }
            catch (Exception ex)
            {
                throw new Exception("AvversitaTrappole.Clona: " + ex.Message);
            }

        }

        public AvversitaTrappole(int code, string description) : base(code,description)
        {
            utilizzi = new List<UtilizzoAvversitaTrappole>();
        }
    }
}
